### PR TITLE
Rename ntp_sponsored_images module to ntp_background_images

### PR DESCRIPTION
**Pull request by simonhong**
This PR doesn't have functional change.

To make review easier https://github.com/brave/brave-core/pull/4866, renaming commit is pulled from it.

Resolves https://github.com/brave/brave-browser/issues/8540

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by simonhong** [comment]
Rebased on C81

File: android/brave_java_sources.gni [hunk 1]
```diff
@@ -51,7 +51,7 @@ brave_java_sources = [
   "../../brave/android/java/org/chromium/chrome/browser/ntp_sponsored_images/SponsoredTab.java",
   "../../brave/android/java/org/chromium/chrome/browser/ntp_sponsored_images/NewTabPageListener.java",
   "../../brave/android/java/org/chromium/chrome/browser/ntp_sponsored_images/RewardsBottomSheetDialogFragment.java",
-  "../../brave/android/java/org/chromium/chrome/browser/ntp_sponsored_images/NTPSponsoredImagesBridge.java",
+  "../../brave/android/java/org/chromium/chrome/browser/ntp_sponsored_images/NTPBackgroundImagesBridge.java",
```
Comment (bridiver) [inline android/brave_java_sources.gni]: are we going to change the java path in another commit? It should be the same for anything in ntp_sponsored_images

**Comment by simonhong** [comment]
I intentionally left most of android side code as-is and only changed c++ and bridge's naming because I don't know much about android implementation and which one should be remained sponsored and which one shouldn't be. Hope @deeppandya will f/u remaining rename issues in android area.

**Comment by simonhong** [comment]
@deeppandya Java file path is also renamed.

**Comment by simonhong** [comment]
Only unrelated `BraveRewardsBrowserTest.TipConnectedPublisherAnonAndConnected` test was failed.

**Pull request merged**